ovs/lib/albaarakoon.py

```diff
@@ -0,0 +1,808 @@
+
+        safety = Configuration.get('/ovs/framework/plugins/alba/config|nsm.safety')
+        AlbaArakoonController._logger.debug('NSM safety is configured at: {0}'.format(safety))
+
+        # Check amount of nodes
+        if len(nsm_cluster.nsm_services) < safety:
+            cls._logger.info('ALBA Backend {0} - Extending if possible'.format(alba_backend.name))
+            current_sr_hosts = [nsm_service.service.storagerouter for nsm_service in nsm_cluster.nsm_services]
+            available_sr_hosts = [storagerouter for storagerouter in nsms_per_storagerouter if storagerouter not in current_sr_hosts]
+            while len(available_sr_hosts) > 0 and len(current_sr_hosts) < safety:
+                candidate_sr = None
+                candidate_load = None
+                for storagerouter in available_sr_hosts:
+                    # Determine the least NSM-loaded Storagerouter to extend to
+                    storagerouter_nsm_load = nsms_per_storagerouter[storagerouter]
+                    if candidate_load is None or storagerouter_nsm_load < candidate_load:
+                        candidate_sr = storagerouter
+                        candidate_load = storagerouter_nsm_load
+                if candidate_sr is None or candidate_load is None:
+                    raise RuntimeError('Could not determine a candidate StorageRouter')
+                current_sr_hosts.append(candidate_sr.ip)
+                available_sr_hosts.remove(candidate_sr)
+                # Extend the cluster (configuration, services, ...)
+                nsm_installer.extend_nsm_cluster(candidate_sr, nsm_cluster)
+
+    @classmethod
+    def ensure_nsm_clusters_load(cls, alba_backend, nsms_per_storagerouter=None, min_internal_nsms=1, external_nsm_cluster_names=None, version_str=None, ssh_clients=None):
+        # type: (AlbaBackend) -> None
+        """
+        Ensure that all NSM clusters are not overloaded
+        :param alba_backend: Alba Backend to ensure NSM Cluster load for
+        :type alba_backend: AlbaBackend
+        :return: None
+        :rtype: NoneType
+        """
+        if ssh_clients is None:
+            ssh_clients = {}
+        if external_nsm_cluster_names is None:
+            external_nsm_cluster_names = []
+
+        nsms_per_storagerouter = nsms_per_storagerouter if nsms_per_storagerouter is not None else cls.get_nsms_per_storagerouter(alba_backend)
+        version_str = version_str or cls.get_alba_version_string()
+        nsm_loads = cls.get_nsm_loads(alba_backend)
+        internal = cls.is_internally_managed(alba_backend)
+        abm_cluster_name = alba_backend.abm_cluster.name
+
+        safety = Configuration.get('/ovs/framework/plugins/alba/config|nsm.safety')
+        maxload = Configuration.get('/ovs/framework/plugins/alba/config|nsm.maxload')
+
+        overloaded = min(nsm_loads.values()) >= maxload
+        if not overloaded:
```

Comment:
logic and what to do is same for if and else. only the loggin differs. so maybe do:
```
if internal:
   nsms_to_add = max(0, min_internal_nsms - len(nsm_loads))
      if not overloaded < log this>
      else <log this>
else:
   nsms_to_add = len(external_nsm_cluster_names)
   <log this or that>
if nsms to add:
   return
```

Reply:
Same number of lines to achieve the same goal